realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -150,25 +150,82 @@ public final boolean isValid() {
     }
 
     /**
-     * Determines if the current RealmObject is obtained synchronously or asynchronously (from a worker thread).
+     * Checks if the query used to find this RealmObject has completed.
+     *
+     * Objects created from async methods like {@link RealmQuery#findFirstAsync()} return a
+     * dummy object that represent the future result of the {@link RealmQuery}. It can be considered
+     * similar to a {@link java.util.concurrent.Future} in this regard.
+     *
+     * Once this method returns {@code true}, this object represents the query result even if the query
+     * didn't find any object matching the query parameters. In this case the RealmObject will
+     * become a "null" object.
```

Comment:
Empty object could also work, neither the term "null object" nor "empty object" is really well-defined, but at least "null" is a concept that is familiar to people. "empty" could mean a lot of things I think...